Marlin/src/lcd/menu/menu_bed_corners.cpp

```diff
@@ -311,21 +319,35 @@ static void _lcd_level_bed_corners_get_next_position() {
     // Select next corner coordinates
     _lcd_level_bed_corners_get_next_position();
 
-    line_to_current_position(manual_feedrate_mm_s.x);
+    do_blocking_move_to(corner_point, manual_feedrate_mm_s.x);
     line_to_z(BED_TRAMMING_HEIGHT);
     if (++bed_corner >= available_points) bed_corner = 0;
   }
 
 #endif // !BED_TRAMMING_USE_PROBE
 
-static void _lcd_level_bed_corners_homing() {
-  _lcd_draw_homing();
+void _lcd_level_bed_corners_homing() {
+  #if HAS_STOWABLE_PROBE && DISABLED(BLTOUCH)
+  if (!all_axes_homed() && probe.deploy) return;
```

Comment:
- The value of `probe.deploy` is the address of the method, and it is always non-zero.
- The `_lcd_level_bed_corners_homing` screen is meant to stay visible until homing is done, as a stepping stone to the next screen handler, so it goes to a new screen handler as soon as homing has been completed. Since it is invoked as a screen handler, it must exit quickly. The `_lcd_test_corners` call sets a new screen handler, but it also runs a loop that takes over for the main loop. It's convoluted, but it works.